packages/hardware-trezor/MASTER_KEY_GENERATION.md

```diff
@@ -0,0 +1,126 @@
+# Trezor Master Key Generation Types
+
+This package now supports multiple master key generation algorithms for Trezor hardware wallets, allowing compatibility with different wallet implementations and enabling seamless integration with various existing wallet types.
+
+## Supported Master Key Generation Types
+
+### 1. ICARUS
+- Standard CIP-3 wallet compatibility mode
+
+### 2. ICARUS_TREZOR
+- CIP-3 variant with 24-word mnemonic compatibility
+- This is Trezor's internal default (SDK passes no `derivationType`)
+
+### 3. LEDGER
+- Ledger hardware wallet compatibility mode
+- Use this if your wallet was originally created on a Ledger device
+- Enables access to Ledger-generated accounts on Trezor hardware
+
+## Key Concept
+
+All types use the same derivation path but different master key generation algorithms from the mnemonic. The `derivationType` tells Trezor which algorithm to use.
+
+## Usage
+
+### Importing Types and Constants
+
+All types and constants are fully exportable from their respective packages:
+
+```typescript
+// Core constants
+import {
+  HD_WALLET_CIP_ID,
+  Cardano
+} from '@cardano-sdk/core';
+
+// Key management types
+import {
+  TrezorConfig,
+  KeyPurpose,
+  MasterKeyGeneration
+} from '@cardano-sdk/key-management';
+
+// Hardware Trezor implementation
+import { TrezorKeyAgent } from '@cardano-sdk/hardware-trezor';
+```
+
+
+
+### Basic Configuration
+
+```typescript
+import { TrezorConfig, MasterKeyGeneration } from '@cardano-sdk/key-management';
+
+const trezorConfig: TrezorConfig = {
+  communicationType: 'web',
+  manifest: {
+    email: 'user@example.com',
+    appUrl: 'https://myapp.com'
+  },
+  // Use CIP-3 master key generation
+  derivationType: 'ICARUS'
+};
+```
+
+
+
+### Creating a Trezor Key Agent
+
+```typescript
+import { TrezorKeyAgent } from '@cardano-sdk/hardware-trezor';
+import { Cardano } from '@cardano-sdk/core';
+import { createBip32Ed25519 } from '@cardano-sdk/crypto';
+
+const dependencies = {
+  logger: console, // or your preferred logger
+  bip32Ed25519: await createBip32Ed25519()
+};
+
+const keyAgent = await TrezorKeyAgent.createWithDevice({
+  chainId: Cardano.ChainIds.Mainnet,
+  accountIndex: 0,
+  trezorConfig: {
+    communicationType: 'web',
+    manifest: {
+      email: 'user@example.com',
+      appUrl: 'https://myapp.com'
+    },
+    derivationType: 'ICARUS'
+  }
+}, dependencies);
+```
+
+
+
+## Discovery & UX Guidelines
```

Comment:
I am not sure if this discovery process is feasible given our current UX. Since we would need to export the root pub key for all three derivation schemes, and execute the discovery on each. Additionally is technically possible for the user to be using more than one derivation scheme and have access on more than one. I think the best/easiest solution here is to simply add the drop down during on-boarding and let the user select which one he wants